File: 0118-pascals-triangle/0118-pascals-triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> ans(numRows);
        
        ans[0] = {1};
        
        if(numRows == 1) return ans;
        
        ans[1] = {1, 1};
        
        if(numRows == 2) return ans;
        
        for(int i=2; i<numRows; i++){
            vector<int> temp = ans[i-1];
            vector<int> newTemp(i+1);
            newTemp[0] = 1;
            for(int j=1; j<i;j++){
                newTemp[j] = temp[j-1] + temp[j];
            }
            newTemp[i] = 1;
            ans[i] = newTemp;
        }
        
        return ans;
        
        
    }
};

File: 0118-pascals-triangle/0118-pascals-triangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        ans = []
        
        ans.append([1])
        
        if numRows==1:
            return ans
        
        ans.append([1,1])
        
        if numRows==2:
            return ans
        
        for i in range(2, numRows):
            temp = [1]
            for j in range(0, len(ans[i-1])-1):
                temp.append(ans[i-1][j]+ans[i-1][j+1])
            temp.append(1)
            ans.append(temp)
        
        
        return ans